### ISSUE TITLE
[[clang::always_destroy]] attribute while emitting exit-time-destructors diagnostic 

### ISSUE THREAD

**Issue by ycdtosa** [llvm/llvm-project#68686]
Compiling this code with the [-Wexit-time-desctuctors](https://clang.llvm.org/docs/DiagnosticsReference.html#wexit-time-destructors) diagnostic enabled will raise the warning, for good reason.

```
static Foo ResourceHandler;
```

here is the output

```
[build] /somefile.cpp:X:14: warning: declaration requires an exit-time destructor [-Wexit-time-destructors]
[build]   static Foo ResourceHandler;
[build]              ^
```
However, being happy with the destructor call on exit time, not only do I want to silence the warning but to flag the variable accordingly.  For that I [add the [[clang::always_destroy]] attribute](https://clang.llvm.org/docs/AttributeReference.html#always-destroy) to the static variable. 

```
[[clang::always_destroy]]
static Foo ResourceHandler;
```

This is meant to tell the compiler that I am aware of the exit time destructor and willing to pay the price, and properly release the resource.

The [[always_destroy]] attibute documentation states that it

> specifies that a variable with static or thread storage duration should have its exit-time destructor run. This attribute is the default unless clang was invoked with -fno-c++-static-destructors.

I believe this is explicit enough to prevent the compiler to emit the diagnostic for that variable, but it is not the case.

**Comment by shafik** [comment]
I think we just need to add a check for `hasAttr<AlwaysDestroyAttr>()` to here: 

https://github.com/llvm/llvm-project/blob/feea5db01360b477b8cf2df03abfa9fc986633d5/clang/lib/Sema/SemaDeclCXX.cpp#L16119-L16121

wdyt @AaronBallman

**Comment by ycdtosa** [comment]
Also `clang/test/SemaCXX/no_destroy.cpp` should be updated.

**Comment by AaronBallman** [comment]
> @AaronBallman. There is not much about [global constructors] (https://clang.llvm.org/docs/DiagnosticsReference.html#wglobal-constructors) diagnostics in the Diagnostics Reference at the moment.

Yeah, it seems this feature was added long ago: https://github.com/llvm/llvm-project/commit/47e40931
And the exit-time destructor was added not long after: https://github.com/llvm/llvm-project/commit/98766db7858379f61d24d5d93b1b7be39fda0913

and there's no review or explanation as to why we wanted to issue the same diagnostic twice here. :-/
